python/lsst/ci/build.py

```diff
@@ -242,35 +280,47 @@ def _build_product(self, product, progress):
 
             fp.write(text)
 
-        # Make executable (equivalent of 'chmod +x $buildscript')
-        st = os.stat(buildscript)
-        os.chmod(buildscript, st.st_mode | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH)
+        retcode = self._execute_build_script(buildscript, logfile, progress, productdir)
+        if not retcode:
```

Comment:
In theory this code could have gone in `_execute_build_script` couldn't it? It's the same below.